### PR TITLE
Calculate and display expected charge during 2PT review - Part 2

### PR DESCRIPTION
**Pull request by Jozzey**
https://eaflood.atlassian.net/browse/WATER-4441

Add functionality to the Review Charge Reference page to allow the user to preview the charge for the Charge Reference.

The preview of the charge will be triggered by a button on the Review Charge Reference details page and will display the result in a banner at the top of the screen.

This is the second of two PRs. In the previous PR https://github.com/DEFRA/water-abstraction-system/pull/1036 a new Charging Module "request" was added to enable a request to be sent to the Charging Module to calculate a charge.

This PR will cover adding the functionality to the screens to enable a charge to be calculated and displayed for a charge reference.

### CODE REVIEW

File: test/support/helpers/review-charge-reference.helper.js [hunk 1]
```diff
@@ -54,9 +54,9 @@ function defaults (data = {}) {
     canalAndRiverTrustAgreement: false,
     twoPartTariffAgreement: true,
     winterDiscount: false,
-    abatementAgreement: 0,
-    chargeAdjustment: 0,
-    amendedChargeAdjustment: 0,
+    abatementAgreement: 1,
```
Comment (Beckyrose200) [inline test/support/helpers/review-charge-reference.helper.js:57]: The comment text at the top of the file needs to be updated for the change to `abatementAgreement`, this currently says 0.

File: app/services/bill-runs/two-part-tariff/calculate-charge.service.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ *
+ * It does this by sending a transaction based on the selected charge reference to the charging module so that it can
+ * calculate the charge. The charge amount is then added to to a flash message which will be displayed to the user.
+ *
+ * @param {String} licenceId - The UUID of the licence related to the charge
+ * @param {String} reviewChargeReferenceId - The UUID of the charge reference review data to calculate the charge on
+ * @param {Object} yar - The Hapi `request.yar` session manager passed on by the controller
+ */
+async function go (licenceId, reviewChargeReferenceId, yar) {
+  const reviewChargeReference = await _fetchReviewChargeReference(reviewChargeReferenceId)
```
[the remark this answers is not in the document]
Comment (Jozzey) [reply app/services/bill-runs/two-part-tariff/calculate-charge.service.js:26]: The other GET routes generally display pages so we have functions to get the data for the page and then another to present the data for the page. This function just calculates a charge and then redirects. If I were to split that out it feels to me like we'd just be doing that for the sake of it. There isn't really much going on in this function so it's not going to make it any clearer. As you pointed out when we had a chat about this. There is another query which is very similar to this one and we could probably share that query between the 2 functions. That may be something we do when we do a refactor when all this work is completed.

File: app/presenters/base.presenter.js [hunk 3]
```diff
@@ -211,6 +211,17 @@ function formatPounds (valueInPence) {
...
 }
 
+/**
+ * Formats a date into a DD/MM/YYYY string, for example, '01/04/2022'
+ *
+ * @param {Date} date The date to be formatted
+ *
+ * @returns {string} The date formatted as a 'DD/MM/YYYY' string
+ */
+function formatShortDate (date) {
```
[the remark this answers is not in the document]
Comment (Jozzey) [reply app/presenters/base.presenter.js:221]: Well spotted. Yes we can use that! I'll remove my formatter and use that one.

File: test/controllers/bill-runs.controller.test.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ const AmendAdjustmentFactorService = require('../../app/services/bill-runs/two-p
 const AmendAuthorisedVolumeService = require('../../app/services/bill-runs/two-part-tariff/amend-authorised-volume.service.js')
 const AmendBillableReturnsService = require('../../app/services/bill-runs/two-part-tariff/amend-billable-returns.service.js')
 const Boom = require('@hapi/boom')
+const CalculateCharge = require('../../app/services/bill-runs/two-part-tariff/calculate-charge.service.js')
```
[the remark this answers is not in the document]
Comment (Jozzey) [reply test/controllers/bill-runs.controller.test.js:16]: Yep, that was a typo

File: test/support/helpers/review-charge-reference.helper.js (hunk 1 above)
Comment (Jozzey) [reply test/support/helpers/review-charge-reference.helper.js:57]: Oops

**Review by Beckyrose200 — APPROVED** [review]

**Pull request merged**